lib/modules/datasource/metadata.ts

```diff
@@ -170,3 +169,41 @@ export function addMetaData(
     }
   }
 }
+
+export function setSourceUrlToHomepage(dep: ReleaseResult): void {
+  if (!dep?.homepage) {
+    return;
+  }
+  const platform = detectPlatform(dep.homepage);
+  if (platform !== 'github' && platform !== 'gitlab') {
+    return;
+  }
+  dep.sourceUrl = dep.homepage;
+  if (shouldDeleteHomepage(dep.sourceUrl, dep.homepage)) {
+    // remove homepage if its not a link to a path in a github/gitlab repo.
+    delete dep.homepage;
+  }
+}
+
+export function shouldDeleteHomepage(
+  sourceUrl: string,
+  homepage: string
+): boolean {
+  const massagedSourceUrl = massageUrl(sourceUrl);
+  if (massagedSourceUrl === homepage) {
+    return true;
+  }
+  const sourceUrlParsed = parseUrl(massagedSourceUrl);
+  if (is.nullOrUndefined(sourceUrlParsed)) {
+    return false;
+  }
+  const homepageParsed = parseUrl(homepage);
+  if (is.nullOrUndefined(homepageParsed)) {
+    return false;
+  }
+  let hPath = homepageParsed.pathname;
+  if (hPath.charAt(hPath.length - 1) === '/') {
+    hPath = hPath.substring(0, hPath.length - 1); // remove last slash
+  }
+  return hPath === sourceUrlParsed.pathname;
```

Comment:
```suggestion
  return trimTrailingSlash(homepage) === trimTrailingSlash(sourceUrl);
```

Don't parse urls

Reply:
If we straightly check without parsing then then tests having `sourceUrl: 'github.com/org/repo'` and `homepage:'www.github.com/org/repo'` will fail. That's why we need to check the path.

But I have removed the manual slash trimming now and used `trimTrailingSlash`